### PR TITLE
Update instructions for building docs

### PR DESCRIPTION
**Pull request by taldcroft**
### Description

This pull request is to clarify the documentation for installing dependencies for building the docs. In particular:
- For conda users, recommend the conda install and explain why that is better. Even though I use conda, when I got to these instructions I just did the first option of `pip install`, thinking they were equivalent.
- Clarify that graphviz is needed or else one gets pages and pages of irrelevant warnings.

### CODE REVIEW

**Comment by mhvk** [comment]
Am in favour of clarifications, but shouldn't `graphviz` be a dependency at least for `pip install -e .[docs]`? cc @astrofrog, @bsipocz

File: docs/install.rst [hunk 1]
```diff
@@ -382,20 +382,21 @@ of this section.
...
+installed.  If you use conda, the easiest way to install the dependencies is
+with::
+
+    conda install -c astropy sphinx-astropy
+
+Without conda, you install the dependencies by specifying ``[docs]`` when
 installing ``astropy`` with pip::
 
-    pip install -e .[docs]
+    pip install -e '.[docs]'
```
Comment (pllim) [inline docs/install.rst:393]: Quotes are not needed.
```suggestion
    pip install -e .[docs]
```

File: docs/install.rst [hunk 2]
```diff
@@ -412,12 +413,16 @@ dependencies, including:
 * `numpydoc <https://numpydoc.readthedocs.io>`_ - an extension to parse
   docstrings in NumPyDoc format
 * `pillow <https://pillow.readthedocs.io>`_ - used in one of the examples
+* `Graphviz <http://www.graphviz.org>`_ - generate inheritance graphs (ONLY
```
Comment (pllim) [inline docs/install.rst:416]: "ONLY" might be too strong a word. Nothing stops you from installing it natively into your system outside of `conda`. See http://www.graphviz.org/download/

File: docs/install.rst [hunk 3]
```diff
@@ -412,12 +413,16 @@ dependencies, including:
...
+  available in conda package)
+
+.. Note::
```
Comment (pllim) [inline docs/install.rst:419]: Does this need to be a separate note? Maybe can be sentences on the Graphviz listing instead?

File: docs/install.rst (hunk 2 above)
Comment (bsipocz) [reply docs/install.rst:416]: agree, maybe just say "not a python package, you need to install it with your preferred system package manager system" or similar. Even if one uses conda for python packages, they may want to homebrew this one, etc.

File: docs/install.rst (hunk 1 above)
Comment (taldcroft) [reply docs/install.rst:393]: Yes, quotes are need in zsh, which is now default on mac.

File: docs/install.rst (hunk 3 above)
Comment (taldcroft) [reply docs/install.rst:419]: I just know that I did not understand this for a while and could not figure out why I was getting thousands of lines of output from building the docs.  So I'm writing these docs based on my former-self experience, which might match others.  Or not?

File: docs/install.rst (hunk 1 above)
Comment (taldcroft) [reply docs/install.rst:393]:
```
(astropy) ➜  astropy git:(enhan-desc-#9660) pip install -e .[docs]
zsh: no matches found: .[docs]
```

Comment (pllim) [reply docs/install.rst:393]: Some people know how I feel about Macs. I am just gonna refrain myself.

Comment (saimn) [reply docs/install.rst:393]: zsh is not a mac thing ;) and at least now they have a modern shell.

Comment (bsipocz) [reply docs/install.rst:393]: what about defining which shell we're assuming? I don't care whether it's bash or zsh or xonsh. (I assume some of the current bash snippets would not work anyway on tcsh which is still the default on the ancient linux distros, etc.)

Comment (taldcroft) [reply docs/install.rst:393]: OT, but zsh has changed my life with oh-my-zsh and having color-coded git branch + dirty tag in the prompt, + branch name autocomplete etc etc.  I know one can get this in other ways, but I never bothered until zsh just made it trivial.

File: docs/install.rst (hunk 2 above)
Comment (taldcroft) [reply docs/install.rst:416]: Agreed, replacing with "generate inheritance graphs (available as a conda package or a system install but not in pip)".

File: docs/install.rst (hunk 1 above)
Comment (bsipocz) [reply docs/install.rst:393]: oh, but I have all of those with bash, too :) (xonsh is the next level though, I saw a few really nice stuff folks do with it, I just never bothered to switch)

**Comment by taldcroft** [comment]
> Reversing the mention of pip vs conda might be controversial. It is like Emacs vs Vim. I am staying out of that one.

About this, I'm not trying to advocate one way or the other, but again reflect my experience as an impatient developer.  When I went to the instructions I typed the first thing I saw that (I thought) would work on my setup, namely the `pip install` command.  But it turned out that does not work because of graphviz.

But turning it around, if a dev is not using conda, then they will necessarily skip down to the pip install because obviously `conda install ..` won't work.  

So the ordering reduces problems without advocating one solution.

**Comment by taldcroft** [comment]
> Nitpick: Only one space after period?

There is no Narrative style guide recommendation on this AFAIK.  I took typing class in high school 36 years ago (most useful class I took) and have been doing 2 spaces ever since.  I have been slowly changing to one in the last year, but unless we have a hard requirement in the style guide I think it can just be whatever.

**Comment by taldcroft** [comment]
> Nitpick: Indent with 4 spaces instead of 2, if we decide to keep the note?

Indentation on notes seems to be random also, there is a 3, 4, and 10 in this RST. I have always used 3 because that is what is in the RST docs (https://docutils.sourceforge.io/docs/ref/rst/directives.html#specific-admonitions), but it looks like it doesn't matter.

**Comment by pllim** [comment]
Re: One or two spaces after period -- I would have to defer to @lglattly on this one.

**Comment by taldcroft** [comment]
I think that one or two spaces doesn't matter in the rendered HTML.  In any case, if there is going to be a policy then all the existing docs need to be brought into conformance and a `spaces-after-periods` check needs to be in CI, otherwise it will inevitably drift.  I honestly think we have bigger problems and the one vs. two spaces should be permanently off the table. People above a certain age tend to use two because they were trained that way.

**Comment by pllim** [comment]
Well, just FYI... https://www.businessinsider.com/microsoft-word-update-2-spaces-after-period-wrong-2020-4

**Comment by taldcroft** [comment]
Fortunately I don't use MS Word for my code editor!

**Comment by lglattly** [comment]
> Re: One or two spaces after period -- I would have to defer to @lglattly on this one.

One space after a period is the editorial standard. (Two spaces were required for typewriters but there's been a slow transition to one space with word processors.) @taldcroft is right that in the HTML two spaces after a period are automatically rendered as one, so it's not a big problem, but I always edit two spaces down to one when copy editing anyway to keep the RST files tidy and consistent. I could add a guideline about it in the style guide if people think that would be helpful!

**Comment by mhvk** [comment]
Hmm, my emacs doesn't even break lines on one space after a period. I think I'm with @taldcroft that if we want to insist on this, we should have some auto-space tool that does it for us.

**Comment by bsipocz** [comment]
> Hmm, my emacs doesn't even break lines on one space after a period

You mean by default? why should it break the line at end of sentence? (I'm certain a lisp macro can be written up though to break it with any character combo)

(disclaimer:  in my native language there isn't double spacing, so seeing it peppered all over the docstrings (but quite inconsistently) puzzled me for a while.)

**Comment by mhvk** [comment]
In `auto-fill-mode`, lines get wrapped at "word breaks" - and the space in `bla. bla` doesn't count (mostly, I suspect, in order to avoid breaking abbreviations).  Anyway, by no means a big deal.  But I find double space after `.` quite helpful in pure text windows; just makes seeing where sentences end easier (and this is, of course, how things are typeset by `TeX` - the space after a period is much wider than one after a regular word).

But likely this is just because, like @taldcroft, I more or less grew up with it... I do think we've spent enough time on it - if we want to insist, have some script replace double spaces...

**Comment by bsipocz** [comment]
Ah, indeed my emacs does that, too and breaks the line before the first 'bla', I never noticed it before. As for TeX, we don't type ligatures either even though TeX typesets the most common combinations by default. 

Anyway, I agree that we have bigger fish to fry than nitpick on this and spend any time on scripting it.

**Pull request merged**